src/StudyViewer/StudyDetails.jsx

```diff
@@ -16,11 +18,27 @@ const onDownload = () => {
     .then(() => message.success('Download has finished', 3));
 };
 
-const onRequestAccess = () => {
-  window.open('https://niaiddevportal.dynamics365portals.us/data-use-request?request_id=123');
-};
-
 class StudyDetails extends React.Component {
+  onRequestAccess = () => {
+    const body = {
+      username: 'TODO remove',
+      resource_path: '/programs/TODO',
+      resource_name: this.props.data.name,
+    };
+    fetchWithCreds({
+      path: `${requestorPath}request`,
+      method: 'POST',
+      body: JSON.stringify(body),
+    }).then(
+      ({ data }) => {
+        if (data && data.redirect_url) {
```

Comment:
do we need to do anything in case of this fails?

Reply:
probably but i'm counting on you to handle it in the real code 😆

Reply:
yeah that's fine, i mean do they (niaid devs) need that when they are playing with this mock up?

Reply:
i don't think so, i told them it's WIP